test/integration/etcd_storage_path_test.go

```diff
@@ -0,0 +1,1249 @@
+var etcdStorageData = map[reflect.Type]struct {
+	ephemeral        bool             // Set to true to skip testing the object
+	stub             runtime.Object   // Valid stub to use during create (this should have at least one field other than name)
+	prerequisites    []runtime.Object // Optional, ordered list of objects to create before stub
+	expectedEtcdPath string           // Expected location of object in etcd, do not use any variables, constants, etc to derive this value - always supply the full raw string
+}{
+	reflect.TypeOf(&oauthapiv1.OAuthClientAuthorization{}): {
+		stub: &oauthapiv1.OAuthClientAuthorization{
+			ClientName: "system:serviceaccount:etcdstoragepathtestnamespace:client",
+			UserName:   "user",
+			UserUID:    "cannot be empty",
+			Scopes:     []string{scope.UserInfo},
+		},
+		prerequisites: []runtime.Object{
+			&kapiv1.ServiceAccount{
+				ObjectMeta: kapiv1.ObjectMeta{
+					Name:        "client",
+					Annotations: map[string]string{saoauth.OAuthRedirectModelAnnotationURIPrefix + "foo": "http://bar"}},
+			},
+			&kapiv1.Secret{
+				ObjectMeta: kapiv1.ObjectMeta{
+					GenerateName: "client",
+					Annotations:  map[string]string{kapi.ServiceAccountNameKey: "client"},
+				},
+				Type: kapiv1.SecretTypeServiceAccountToken,
+			},
+		},
+		expectedEtcdPath: "openshift.io/oauth/clientauthorizations/user:system:serviceaccount:etcdstoragepathtestnamespace:client",
+	},
+	reflect.TypeOf(&oauthapiv1.OAuthClient{}): {
+		stub: &oauthapiv1.OAuthClient{
+			ObjectMeta: kapiv1.ObjectMeta{Name: "client"},
+		},
+		expectedEtcdPath: "openshift.io/oauth/clients/client",
+	},
+	reflect.TypeOf(&oauthapiv1.OAuthAuthorizeToken{}): {
+		stub: &oauthapiv1.OAuthAuthorizeToken{
+			ObjectMeta: kapiv1.ObjectMeta{Name: "tokenneedstobelongenoughelseitwontwork"},
+			ClientName: "client0",
+			UserName:   "user",
+			UserUID:    "cannot be empty",
+		},
+		prerequisites: []runtime.Object{
+			&oauthapiv1.OAuthClient{
+				ObjectMeta: kapiv1.ObjectMeta{Name: "client0"},
+			},
+		},
+		expectedEtcdPath: "openshift.io/oauth/authorizetokens/tokenneedstobelongenoughelseitwontwork",
+	},
+	reflect.TypeOf(&oauthapiv1.OAuthAccessToken{}): {
+		stub: &oauthapiv1.OAuthAccessToken{
+			ObjectMeta: kapiv1.ObjectMeta{Name: "tokenneedstobelongenoughelseitwontwork"},
+			ClientName: "client1",
+			UserName:   "user",
+			UserUID:    "cannot be empty",
+		},
+		prerequisites: []runtime.Object{
+			&oauthapiv1.OAuthClient{
+				ObjectMeta: kapiv1.ObjectMeta{Name: "client1"},
+			},
+		},
+		expectedEtcdPath: "openshift.io/oauth/accesstokens/tokenneedstobelongenoughelseitwontwork",
+	},
+	reflect.TypeOf(&oauthapiv1.OAuthRedirectReference{}): {ephemeral: true}, // Used for specifying redirects, never stored in etcd
+
+	reflect.TypeOf(&imageapidockerpre012.DockerImage{}): {ephemeral: true}, // Not a real object so cannot be stored in etcd  // TODO confirm this
```

Comment:
@soltysh everything is already organized by import path.  Not sure if we need / want a different organization since keeping imports together is easy.